src/integrate/scvi_tools/script.py

```diff
@@ -0,0 +1,215 @@
+            adata_reference,
+            **arches_params,
+            n_hidden=128, #this is the default
+            n_latent=30,
+            n_layers=2,
+            dropout_rate=0.1, #this is the default
+            dispersion='gene', #this is the default
+            gene_likelihood='nb',
+            encode_covariates=True, #Parameterization for better scArches performance -> maybe don't use this always?
+            deeply_inject_covariates=False, #Parameterization for better scArches performance -> maybe don't use this always?
+            use_observed_lib_size=False, #When size_factors are not passed
+        )
+
+        vae_reference.train(
+            max_epochs=par['max_epochs'],
+            early_stopping=par['early_stopping'],
+            early_stopping_monitor=par['early_stopping_monitor'],
+            early_stopping_patience=par['early_stopping_patience'],
+            early_stopping_min_delta=par['early_stopping_min_delta'],
+            plan_kwargs=PLAN_KWARGS,
+            check_val_every_n_epoch=1,
+            use_gpu=(cuda_is_available() or mps_is_available()),
+        )
+
+        if par["base_model"] == "scvi":
+            model = scvi.model.SCVI
+
+        elif par["base_model"] == "scanvi":
+            vae_reference = scvi.model.SCANVI.from_scvi_model(
+                vae_reference,
+                unlabeled_category=par["unlabeled_category"],
+                labels_key=par["labels_key"],
+            )
+
+            vae_reference.train(max_epochs=20, n_samples_per_label=100)
+            model = scvi.model.SCANVI
+
+    elif par["base_model"] == "totalvi":
+
+        if par["reference_proteins_key"] in mdata_reference.mod:
+            # Put the proteins modality into reference adata
+            proteins_reference = mdata_reference.mod[par["reference_proteins_key"]]
+            adata_reference.obsm[par["reference_proteins_key"]] = proteins_reference
+
+        scvi.model.TOTALVI.setup_anndata(
+            adata_reference,
+            batch_key=par['obs_batch'],
+            protein_expression_obsm_key=par["reference_proteins_key"]
+        )
+
+        vae_reference = scvi.model.TOTALVI(
+            adata_reference,
+            **arches_params
+        )
+        vae_reference.train()
+
+        model = scvi.model.TOTALVI
+
+    return adata_reference, vae_reference, model
+
+
+def map_query_to_new_reference(mdata_reference, mdata_query, adata_query):
+    """Build model on the provided reference and map query to the reference"""
+
+    if par["var_input"]:
+        # Subset to HVG
+        adata_query = adata_query[:,adata_query.var["var_input"]].copy()
+
+    adata_reference, vae_reference, model = build_referrence_model(mdata_reference)
+
+    if par["base_model"] == "totalvi":
+        adata_query = align_proteins_names(adata_reference, mdata_query, adata_query)
+        
+    # Reorder genes and pad missing genes with 0s
+    model.prepare_query_anndata(adata_query, vae_reference)
```

Comment:
I am getting an error here when running the tests:

```python3
AttributeError: type object 'SCVI' has no attribute 'prepare_query_anndata'
```

Reply:
Done, should work now